### PR TITLE
Re-add TZ assertion and fix missing dtbo.img

### PR DESCRIPTION
**Pull request by ivecera**
Recent commit removed recovery updater library and release_tools.py script because the TZ based assertion was not functional. Side effect of this change was that dtbo.img is not part of flashable ZIP.

This series re-adds back this functionality and fixes it so it can be used on Xiaomi sdm710 based devices. To use the assertion create ```board-info.txt``` file in the device tree with the following content:

```require version-trustzone=<TZ_VERSION>```

For pyxis I'm using ```XF.5.0.2.C5-00006``` for LineageOS 17.1. Note that you must not to include ```TZ.``` prefix to TZ_VERSION string.

### CODE REVIEW

**Review by NaassomDourado — APPROVED** [review]

File: recovery/recovery_updater.cpp [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+#include <vector>
+
+#include "edify/expr.h"
+#include "otautil/error_code.h"
+
+#define MAX(a, b) (((a) > (b)) ? (a) : (b))
+
+#define ALPHABET_LEN 256
+
+#define TZ_PART_PATH "/dev/block/bootdevice/by-name/tz"
```
Comment (uixdess) [inline recovery/recovery_updater.cpp:39]:
```suggestion
```

File: recovery/recovery_updater.cpp [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+#define TZ_VER_STR "QC_IMAGE_VERSION_STRING=TZ."
```
Comment (uixdess) [inline recovery/recovery_updater.cpp:40]:
```suggestion
```

File: recovery/recovery_updater.cpp [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+err_ret:
+    return ret;
+}
+
+/* verify_trustzone("TZ_VERSION", "TZ_VERSION", ...) */
+Value* VerifyTrustZoneFn(const char* name, State* state,
+                     const std::vector<std::unique_ptr<Expr>>& argv) {
+    char current_tz_version[TZ_VER_BUF_LEN];
+    int ret;
+
```
Comment (uixdess) [inline recovery/recovery_updater.cpp:167]:
```suggestion
        char* TZ_PART_PATH = strdup("/dev/block/bootdevice/by-name/tz");
        char* TZ_VER_STR = strdup("QC_IMAGE_VERSION_STRING=TZ.");

```

**Review by uixdess — CHANGES_REQUESTED** [review]
@ivecera let's remove annoying C++ warning
`warning: ISO C++11 does not allow conversion from string literal to 'char *'`

**Review by uixdess — APPROVED** [review]

**Pull request merged**